File: python-client-sdk/src/sdk/models/shared/codat_datacontracts_datasets_weblink.py
from __future__ import annotations
import dataclasses
from ..shared import codat_datacontracts_datasets_weblinktype_enum as shared_codat_datacontracts_datasets_weblinktype_enum
from dataclasses_json import Undefined, dataclass_json
from sdk import utils
from typing import Optional


@dataclass_json(undefined=Undefined.EXCLUDE)
@dataclasses.dataclass
class CodatDataContractsDatasetsWebLink:
    type: shared_codat_datacontracts_datasets_weblinktype_enum.CodatDataContractsDatasetsWebLinkTypeEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    url: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('url'), 'exclude': lambda f: f is None }})
    

File: typescript-client-sdk/src/sdk/models/shared/codatdatacontractsdatasetsweblink.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatDataContractsDatasetsWebLinkTypeEnum } from "./codatdatacontractsdatasetsweblinktypeenum";
import { Expose } from "class-transformer";


export class CodatDataContractsDatasetsWebLink extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "type" })
  type: CodatDataContractsDatasetsWebLinkTypeEnum;

  @SpeakeasyMetadata()
  @Expose({ name: "url" })
  url?: string;
}